src/java.base/share/classes/jdk/internal/foreign/abi/SharedUtils.java

```diff
@@ -260,6 +261,13 @@ static void handleUncaughtException(Throwable t) {
         }
     }
 
+    static long unboxSegment(MemorySegment segment) {
+        if (segment instanceof HeapMemorySegmentImpl) {
```

Comment:
there's MemorySegment.isNative, which maybe is better?

Reply:
Yeah, that works. It also includes mapped segments